src/components/fx/attributes.js

```diff
@@ -50,6 +50,17 @@ module.exports = {
                 '`namelength - 3` characters and add an ellipsis.'
             ].join(' ')
         },
+        zformat: {
```

Comment:
We can't declare `zformat` in `fx/` as it only makes sense for traces with `z` data array (e.g. `heatmap`, `contour` and `surface`).

Reply:
... we wouldn't want traces w/o a `z` attribute to have a `zformat` under their name on https://plot.ly/javascript/reference/

Reply:
... this has me thinking - maybe my `hoverlabel.zformat` idea of https://github.com/plotly/plotly.js/issues/1383#issuecomment-329229633 was wrong. It might be best to keep the `hoverlabel` attribute container _as is_ that way that set of attributes will be the same for all trace types.

Reply:
So maybe `zhoverformat` is the best option. I hate that it doesn't extend to other trace types very well. But I prefer this over breaking `hoverlabel` symmetry.

Any thoughts on this @alexcjohnson @cldougl :question:

Reply:
> So maybe `zhoverformat` is the best option. I hate that it doesn't extend to other trace types very well. But I prefer this over breaking `hoverlabel` symmetry.

I suppose the only other solution would be to move `hoverlabel` into each trace type `attributes` so it can be different for each one. But that's both a much more invasive change, and you're right there's something nice about keeping that container consistent in structure (though a lot of other containers, like `marker` and `line`, DO change from type to type... so it's not without precedent)

Whatever solution we adopt we should think about how it will extend to `intensity` and `surfacecolor`
https://github.com/plotly/plotly.js/issues/1383#issuecomment-280963522 - `surfacecolorhoverformat` is getting a bit unwieldy but... so is `hoverlabel.surfacecolorformat`

So yeah, I guess `zhoverformat` is sounding better and better 😅

Reply:
+1 for keeping `hoverlabel` symmetry at this point.